phantom_connector.py

```diff
@@ -562,22 +563,38 @@ def _find_artifacts(self, param):
         url_enc_values = quote(values, safe='')
 
         if cef_key and exact_match:
-            endpoint = '/rest/artifact?_filter_cef__{}={}&page_size=0&pretty'.format(quote(cef_key, safe=''), repr(url_enc_values))
+            endpoint = f"/rest/artifact?_filter_cef__{quote(cef_key, safe='')}={repr(url_enc_values)}&pretty"
         elif cef_key:
-            endpoint = '/rest/artifact?_filter_cef__{}__{}={}&page_size=0&pretty'.format(quote(cef_key, safe=''),
-                                                                                        "icontains", repr(url_enc_values))
+            endpoint = f"/rest/artifact?_filter_cef__{quote(cef_key, safe='')}__icontains={repr(url_enc_values)}&pretty"
         else:
-            endpoint = '/rest/artifact?_filter_cef__{}={}&page_size=0&pretty'.format("icontains", repr(url_enc_values))
+            endpoint = f"/rest/artifact?_filter_cef__icontains={repr(url_enc_values)}&pretty"
 
         if limit_search:
             endpoint += '&_filter_container__in={}'.format(container_ids)
 
-        ret_val, response, resp_data = self._make_rest_call(endpoint, action_result)
+        records = list()
+        page = 0
+        page_size = 10
+        while True:
+            count = (page+1)*10
+            if max_results != 0 and count > max_results:
```

Comment:
What should be the behaviour if we set `max_results` to less than 10? Right now, as far as I see, the connector will fetch 10 artefacts even if we set `max_results` to less than 10

Reply:
I have executed the code and verified this and if we put any number below 10, it adheres whatever the input that has been provided.